src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -365,7 +366,15 @@ internal int GetHashCodeNoParameters()
         internal override DiagnosticInfo? GetUseSiteDiagnostic()
         {
             DiagnosticInfo? info = null;
-            CalculateUseSiteDiagnostic(ref info);
+            if (CallingConvention.IsCallingConvention(CallingConvention.ExtraArguments) ||
+                CallingConvention.IsCallingConvention(CallingConvention.FastCall))
+            {
+                info = new CSDiagnosticInfo(ErrorCode.ERR_UnsupportedCallingConvention, ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat));
```

Comment:
>ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat) [](start = 88, length = 61)

This looks like a premature string allocation, we do not format symbols until absolutely necessary. If default formatting doesn't work, we have a FormattedSymbol helper type, or something like that.